src/Integration.Vsix/Settings/SonarLintSettings.cs

```diff
@@ -23,44 +23,31 @@
 using Microsoft.VisualStudio.Settings;
 using Microsoft.VisualStudio.Shell;
 using Microsoft.VisualStudio.Shell.Interop;
-using Microsoft.VisualStudio.Shell.Settings;
+using SonarLint.VisualStudio.Integration.Vsix.Settings;
 
 namespace SonarLint.VisualStudio.Integration.Vsix
 {
     [Export(typeof(ISonarLintSettings))]
     [PartCreationPolicy(CreationPolicy.Shared)]
-    public class SonarLintSettings : ISonarLintSettings, IProfileManager
+    internal class SonarLintSettings : ISonarLintSettings, IProfileManager
     {
         public const string SettingsRoot = "SonarLintForVisualStudio";
 
-        private readonly WritableSettingsStore writableSettingsStore;
+        // Lazily create the settings store on first use (we can't create it in the constructor)
+        private readonly Lazy<WritableSettingsStore> writableSettingsStore;
 
         [ImportingConstructor]
-        public SonarLintSettings(SVsServiceProvider serviceProvider)
-            : this(new ShellSettingsManager(serviceProvider))
```

Comment:
Previously the class created a concrete VS class in its constructor.
I've made two related changes:
1. the code that creates the VS class has been moved to a separate class, and
2. I added a `Lazy` property to this class to delay fetching the VS service until it is required.